futures-util/src/future/shared.rs

```diff
@@ -181,7 +198,7 @@ impl<Fut: Future> Future for Shared<Fut> {
                 return Poll::Pending
             }
             COMPLETE => {
-                return unsafe { Poll::Ready(this.clone_output()) };
+                return unsafe { Poll::Ready(this.take_output()) };
```

Comment:
That's too drastic: *Every* instance of `Shared` needs produce the same `Output`. `Shared`'s main purpose is enable sharing of an eventual value that hasn't been produced yet.

Reply:
I suppose I didn't edit my comment above quickly enough ^^'